File: src/incident_intel/schemas/conversation.py
"""Pydantic schema for conversations."""

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from incident_intel.models.conversation import MessageRole
from incident_intel.schemas.chat import SourceItem


class MessageResponse(BaseModel):
    """Response schema for a single message in conversation."""

    id: UUID
    role: Literal[MessageRole.USER, MessageRole.ASSISTANT]  # excludes SYSTEM
    content: str
    created_at: datetime
    sources: list[SourceItem] = []

    model_config = ConfigDict(from_attributes=True)


class ConversationResponse(BaseModel):
    """Response schema for conversation metadata."""

    id: UUID
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ConversationDetailResponse(ConversationResponse):
    """Response schema for full conversation details.

    Includes list of messages for a single conversation.
    """

    messages: list[MessageResponse]


class ConversationListResponse(BaseModel):
    """Reponse schema for conversation list."""

    items: list[ConversationResponse]
    total: int = Field(..., description="Total number of conversations")

File: ui/src/schemas/api/conversation.ts
import { z } from "zod";

export const sourceItemSchema = z.object({
  chunk_id: z.string(),
  document_id: z.string(),
  document_title: z.string(),
  chunk_index: z.number(),
  relevance_score: z.number(),
});

export const messageSchema = z.object({
  id: z.string(),
  role: z.enum(["user", "assistant"]),
  content: z.string(),
  created_at: z.string(),
  sources: z.array(sourceItemSchema),
});

export const conversationSchema = z.object({
  id: z.string(),
  created_at: z.string(),
});

export const conversationDetailSchema = z.object({
  id: z.string(),
  created_at: z.string(),
  messages: z.array(messageSchema),
});

export const conversationListSchema = z.object({
  items: z.array(conversationSchema),
  total: z.number(),
});
